backend/src/main/scala/sbt/internal/inc/bloop/internal/BloopNameHashing.scala

```diff
@@ -168,21 +167,22 @@ private final class BloopNameHashing(
       val previous = previousAnalysis.stamps
       val previousRelations = previousAnalysis.relations
 
-      val hashesMap = uniqueInputs.sources.map(kv => kv.source -> kv.hash).toMap
       val sourceChanges = tracer.traceVerbose("source changes") { _ =>
         lookup.changedSources(previousAnalysis).getOrElse {
           val previousSources = previous.allSources.toSet
           new UnderlyingChanges[VirtualFileRef] {
             private val sourceRefs = sources.map(f => f: VirtualFileRef)
-            private val inBoth = previousSources & sourceRefs
-            val removed = previousSources -- inBoth
-            val added = sourceRefs -- inBoth
+            private val inBoth = sourceRefs.filter(previousSources)
```

Comment:
Otherwise we get PlainFile from previous sources and we want the current sources